tf_quant_finance/math/jacobian.py

```diff
@@ -0,0 +1,120 @@
+# Lint as: python3
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Helper functions for computing jacobian."""
+
+
+import functools
+
+import tensorflow.compat.v2 as tf
+
+
+def jacobian(func, xs, unconnected_gradients=None, name=None,
+             parallel_iterations=None, experimental_use_pfor=True):
+  """Computes the jacobian of `func` wrt to `xs`.
+  Args:
+    func: Python callable accepting one `Tensor` of shape of `x` and returning
+      a `Tensor` of any shape. The function whose jacobian is to be computed.
```

Comment:
Replace `x` with `xs`? Please see the comment below.

Reply:
As discussed before, addressed in 8a357ce